ast/ast.go

```diff
@@ -202,18 +200,27 @@ func (il *IntegerLiteral) String() string {
 	return strconv.FormatInt(il.Value, 10)
 }
 
+type BooleanType int
+
+const (
+	False BooleanType = iota
+	True
+)
+
+var BooleanTypeMap = map[BooleanType]string{
```

Comment:
Why did you create `BooleanTypeMap` without using string `true` or `false`?

Reply:
At first thought this map is for keeping consistency with data structure or operator. And booleanLiteral `String()` method become much simple.

But on second thought, this can be style issue :) what do you think? which one looks better?

Reply:
It is just my opinion, 'int' and 'string', 'boolean' is same context.
We don't create IntType, StringType and we don't implement those struct in IntegerLiteral, StringLiteral.
These, only have their value. So, why does booleanLiteral have their type value?
I think it would be more complicated.